server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetaData.java

```diff
@@ -0,0 +1,291 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster.coordination;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CoordinationMetaData implements Writeable, ToXContentFragment {
+
+    public static final CoordinationMetaData EMPTY_META_DATA = builder().build();
+
+    private final long term;
+
+    private final VotingConfiguration lastCommittedConfiguration;
+
+    private final VotingConfiguration lastAcceptedConfiguration;
+
+    private final Set<DiscoveryNode> votingTombstones;
+
+    private static final ParseField TERM_PARSE_FIELD = new ParseField("term");
+    private static final ParseField LAST_COMMITTED_CONFIGURATION_FIELD = new ParseField("last_committed_config");
+    private static final ParseField LAST_ACCEPTED_CONFIGURATION_FIELD = new ParseField("last_accepted_config");
+
+    private static long term(Object[] termAndConfigs) {
+        return (long)termAndConfigs[0];
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastCommittedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[1];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastAcceptedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[2];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    private static final ConstructingObjectParser<CoordinationMetaData, Void> PARSER = new ConstructingObjectParser<>(
+            "coordination",
```

Comment:
how about calling this coordination_metadata?

Reply:
https://github.com/elastic/elasticsearch/pull/35625/commits/d99c52d27586ef04a4ce64d3ce4daf6ef471cf73